### ISSUE TITLE
Add WAR distribution

### ISSUE THREAD

**Issue by olegshulyakov** [metabase/metabase#4407]
### Feature requests and proposals
Propose to add WAR distribution to make Metabase be installed into different application servers.

Here is the link on Stackoverflow: [How to  #create WAR](http://stackoverflow.com/questions/33135667/can-i-make-lein-ring-server-headless-run-on-a-specific-servlet-context)

**Comment by camsaul** [comment]
Hi @olegshulyakov, if you have a sense of what needs to be done to add this, we'd love to have you as a contributor and would welcome a PR to support this 👍

**Comment by olegshulyakov** [comment]
Hi @camsaul I have forked the repository and invetigate how it can be done.
But I`m new with Clojure and JavaScript, I am more familier with Java.

**Comment by duck1123** [comment]
This might be helpful. https://github.com/luminus-framework/lein-uberwar

**Comment by jcode** [comment]
+1

**Comment by camsaul** [comment]
Hey everyone, one way we prioritize issues is by sorting them by the number of 👍 reactions on each issue's description. So instead of leaving `+1` or 👍  comments, please just upvote by adding a reaction to the issue description.

**Comment by dchumpitaz** [comment]
I'm very interested too. For install in WildFly / JBoss.

**Comment by jainmukesh** [comment]
I am interested in war distribution to install on Jboss / Wildfly.

**Comment by aadrian** [comment]
We are also interested in a WAR distribution, in order to be able to deploy it into existing Application Server infrastructure (Tomcat, TomEE, WebLogic, Websphere, etc.) - this would greatly ease the adoption of MetaBase.

Since MetaBase is already using Ring, than this Lein plug-in https://github.com/weavejester/lein-ring#war-files  should be able to do the job of creating a WAR file.

**Comment by ananthu2050** [comment]
I would also love to have this issue addressed👍

**Comment by radi-ratlh** [comment]
It's very important to have a war file distribution, because some environments have restrictions on copying and executing new files.

**Comment by devMls** [comment]
+1 corporate enviroments only accept this classical distribution mode. In addition a war is a handly way to integrate metabase in complex scenarios,.

**Comment by cabintech** [comment]
Yes this would be very useful for us. We have a web server infrastructure for both external and internal use (Tomcat based). It is well vetted and integrated into our cloud platform. Would really would like to just drop a JAR into Tomcat and use all our existing server management infrastructure, including password/user management, SSL/HTTPS, data sources for DB connection management, etc. Really don't want another server architecture (Jetty) to learn and manage.

**Comment by psarin** [comment]
+1 vote for this feature. We would love to deploy it into our Tomcat server environment.
